core/src/main/java/org/apache/iceberg/rest/requests/UpdateRequirementParser.java

```diff
@@ -0,0 +1,264 @@
+        break;
+      case ASSERT_CURRENT_SCHEMA_ID:
+        writeAssertCurrentSchemaId((UpdateRequirement.AssertCurrentSchemaID) updateRequirement, generator);
+        break;
+      case ASSERT_DEFAULT_SPEC_ID:
+        writeAssertDefaultSpecId((UpdateRequirement.AssertDefaultSpecID) updateRequirement, generator);
+        break;
+      case ASSERT_DEFAULT_SORT_ORDER_ID:
+        writeAssertDefaultSortOrderId((UpdateRequirement.AssertDefaultSortOrderID) updateRequirement, generator);
+        break;
+      default:
+        throw new IllegalArgumentException(
+            String.format("Cannot convert update requirement to json. Unrecognized type: %s", requirementType));
+    }
+
+    generator.writeEndObject();
+  }
+
+  /**
+   * Read MetadataUpdate from a JSON string.
+   *
+   * @param json a JSON string of a MetadataUpdate
+   * @return a MetadataUpdate object
+   */
+  public static UpdateRequirement fromJson(String json) {
+    try {
+      return fromJson(JsonUtil.mapper().readValue(json, JsonNode.class));
+    } catch (IOException e) {
+      throw new UncheckedIOException("Failed to read JSON string: " + json, e);
+    }
+  }
+
+  public static UpdateRequirement fromJson(JsonNode jsonNode) {
+    Preconditions.checkArgument(jsonNode != null && jsonNode.isObject(),
+        "Cannot parse update requirement from non-object value: %s", jsonNode);
+    Preconditions.checkArgument(jsonNode.hasNonNull(TYPE), "Cannot parse update requirement. Missing field: type");
+    String type = JsonUtil.getString(TYPE, jsonNode).toLowerCase(Locale.ROOT);
+
+    switch (type) {
+      case ASSERT_TABLE_DOES_NOT_EXIST:
+        return readAssertTableDoesNotExist(jsonNode);
+      case ASSERT_TABLE_UUID:
+        return readAssertTableUUID(jsonNode);
+      case ASSERT_REF_SNAPSHOT_ID:
+        return readAssertRefSnapshotId(jsonNode);
+      case ASSERT_LAST_ASSIGNED_FIELD_ID:
+        return readAssertLastAssignedFieldId(jsonNode);
+      case ASSERT_LAST_ASSIGNED_PARTITION_ID:
+        return readAssertLastAssignedPartitionId(jsonNode);
+      case ASSERT_CURRENT_SCHEMA_ID:
+        return readAssertCurrentSchemaId(jsonNode);
+      case ASSERT_DEFAULT_SPEC_ID:
+        return readAssertDefaultSpecId(jsonNode);
+      case ASSERT_DEFAULT_SORT_ORDER_ID:
+        return readAssertDefaultSortOrderId(jsonNode);
+      default:
+        throw new UnsupportedOperationException(
+            String.format("Unrecognized update requirement. Cannot convert to json: %s", type));
+    }
+  }
+
+  private static void writeAssertTableUUID(UpdateRequirement.AssertTableUUID requirement, JsonGenerator gen)
+      throws IOException {
+    gen.writeStringField(UUID, requirement.uuid());
+  }
+
+  private static void writeAssertRefSnapshotId(UpdateRequirement.AssertRefSnapshotID requirement, JsonGenerator gen)
+      throws IOException {
+    gen.writeStringField(NAME, requirement.refName());
+    gen.writeNumberField(SNAPSHOT_ID, requirement.snapshotId());
```

Comment:
I think this should test whether the required snapshot ID is null and explicitly write `null` if it is.

Reply:
(This is the only one that can be null)

Reply:
I though this would write `null` if the `Long` value were null. I'll write a test to ensure it though.

Reply:
You're right. `writeNumberField` doesn't handle null properly.

Updated and added a test.